perun-core/src/main/java/cz/metacentrum/perun/core/impl/ServicesManagerImpl.java

```diff
@@ -192,6 +195,115 @@ public ServicesManagerImpl(DataSource perunPool) {
 		return new RichDestination(destination, facility, service);
 	};
 
+	@SuppressWarnings("ConstantConditions")
+	@Override
+	public void blockServiceOnFacility(int serviceId, int facilityId) throws InternalErrorException, ServiceAlreadyBannedException {
+		int newBanId = Utils.getNewId(jdbc, "service_denials_id_seq");
+		try {
+			// jdbc template cannot be null
+			jdbc.update("insert into service_denials(id, facility_id, service_id) values (?,?,?)", newBanId, facilityId, serviceId);
+		} catch (DuplicateKeyException ex) {
+			throw new ServiceAlreadyBannedException(String.format("Service with id %d is already banned on the facility with id %d", serviceId, facilityId));
+		}
+	}
+
+	@SuppressWarnings("ConstantConditions")
+	@Override
+	public void blockServiceOnDestination(int serviceId, int destinationId) throws InternalErrorException, ServiceAlreadyBannedException {
+		try {
+			int newBanId = Utils.getNewId(jdbc, "service_denials_id_seq");
+			// jdbc template cannot be null
+			jdbc.update("insert into service_denials(id, destination_id, service_id) values (?,?,?)", newBanId, destinationId, serviceId);
+		} catch (DuplicateKeyException ex) {
+			throw new ServiceAlreadyBannedException(String.format("Service with id %d is already banned on the destination with id %d", serviceId, destinationId));
+		}
+	}
+
+	@SuppressWarnings("ConstantConditions")
+	@Override
+	public List<Service> getServicesBlockedOnFacility(int facilityId) {
+		// jdbc template cannot be null
+		return jdbc
+			.query("select " + ServicesManagerImpl.serviceMappingSelectQuery +
+					" from services left join service_denials on service_denials.service_id = services.id where service_denials.facility_id = ?",
+				ServicesManagerImpl.SERVICE_MAPPER, facilityId);
+	}
+
+	@SuppressWarnings("ConstantConditions")
+	@Override
+	public List<Service> getServicesBlockedOnDestination(int destinationId) {
+		// jdbc template cannot be null
+		return jdbc
```

Comment:
Please wrap all calls of jdbc in try/catch like we do in other methods of ``ServicesManagerImpl``.

```java
try {
  // jdbc.query() or jdbc.update() logic
} catch(RuntimeException ex) {
	throw new InternalErrorException(ex);
}
```
We want to throw our exception and not sql or other runtime exceptions outside of Perun. If you already catch some exception, than add this catch block as last, so that more specific exceptions are handled by their own catch block and rest of them falls back to this generic block.